### PR TITLE
add array support for metadata, fixes #4

### LINKED ISSUE #4: allow arrays in metadata
What a great plug-in, thanks!

> Metadata values must be scalar (bool|float|int|string).

Can you allow arrays, since PHP supports array constants?  I found myself needing to associate a list of possible tags with a class..

```php
#[Metadata('actions', 'list,check,clean')]

class MyClass {
    const ACTIONS=['list,'check','clean']; // replace this with metadata call
```

Of course, I guess another approach would be Enums, which I should probably use more than I do...

There's probably a good reason why these are scalar, but I figured I'd ask the question anyway.

### PR DESCRIPTION
**Pull request by tacman**
I updated the docs, attribute and tests.

It got me thinking, though, if null should be allowed as a valid metadata value.

Nonetheless, I think this fixes #4

### CODE REVIEW

**Comment by tacman** [comment]
I fixed the phpstan errors, except for one:

```bash
vendor/bin/phpstan
```

> ::error file=src/Metadata.php,line=22,col=0::Method Zenstruck\Metadata::__construct() has parameter $value with no value type specified in iterable type array.


```php
final class Metadata
{
    public function __construct(
        /** @readonly */
        public string $key,

        /**
         * @readonly
         * @param string|bool|int|float|mixed value
         */
        public string|bool|int|float|array $value,
    ) {
    }
```

What should this be?

File: src/Metadata.php [hunk 1]
```diff
@@ -23,8 +23,11 @@ public function __construct(
         /** @readonly */
         public string $key,
 
-        /** @readonly */
-        public string|bool|int|float $value,
+        /**
+         * @readonly
+         * @param string|bool|int|float|mixed value
```
Comment (kbond) [inline src/Metadata.php]:
```suggestion
         * @param scalar|array<scalar> value
```

File: src/Metadata.php [hunk 2]
```diff
@@ -45,8 +48,9 @@ public static function for(object|string $objectOrClassOrAlias): array
      * or null if key does not exist.
      *
      * @param object|class-string|string $objectOrClassOrAlias
+     * @return string|bool|int|float|array<mixed> value
```
Comment (kbond) [inline src/Metadata.php]:
```suggestion
     * @return scalar|array<scalar>|null
```

File: src/Metadata.php [hunk 3]
```diff
@@ -56,8 +60,9 @@ public static function get(object|string $objectOrClassOrAlias, string $key): st
      * class/object/alias or null if no keys exist.
      *
      * @param object|class-string|string $objectOrClassOrAlias
+     * @return  string|bool|int|float|array<mixed>|null value
```
Comment (kbond) [inline src/Metadata.php]:
```suggestion
     * @return scalar|array<scalar>|null
```

File: src/Metadata/Map.php [hunk 4]
```diff
@@ -27,7 +27,7 @@ final class Map
     /** @var array<string,class-string> */
     public array $aliasToClassMap = [];
 
-    /** @var array<string|class-string,array<string,scalar>> */
+    /** @var array<string|class-string,array<string,scalar|mixed[]>> */
```
Comment (kbond) [inline src/Metadata/Map.php]:
```suggestion
    /** @var array<string|class-string,array<string,scalar|array<scalar>>> */
```

**Review by kbond — CHANGES_REQUESTED** [review]
Looks good! I think my suggestions will fix phpstan.

> It got me thinking, though, if null should be allowed as a valid metadata value.

Couldn't hurt but maybe wait until someone needs it.

**Comment by kbond** [comment]
I'll take a look at fixing phpstan before merging.

**Review by kbond — APPROVED** [review]

**Comment by kbond** [comment]
Thanks @tacman!

**Pull request merged**